src/tools/add_entity.cc

```diff
@@ -68,7 +68,7 @@ struct add_entity_tool : tool {
         return place->rot;
     }
 
-    void use() override {
+    void use(ship_space *ship) override {
```

Comment:
This doesn't seem right.

1/ The point you're most likely to need to interact with the ship is pre_use.
2/ Tools already have access via the global ship.